crates/sui-graphql-rpc/src/metrics.rs

```diff
@@ -1,15 +1,209 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use prometheus::{register_histogram_with_registry, Histogram, Registry};
-
-#[derive(Clone, Debug)]
-pub struct RequestMetrics {
-    pub(crate) input_nodes: Histogram,
-    pub(crate) output_nodes: Histogram,
-    pub(crate) query_depth: Histogram,
-    pub(crate) query_payload_size: Histogram,
-    pub(crate) _db_query_cost: Histogram,
+use std::sync::Arc;
+
+use async_graphql::{PathSegment, ServerError};
+use prometheus::{
+    register_histogram_vec_with_registry, register_histogram_with_registry,
+    register_int_counter_vec_with_registry, Histogram, HistogramVec, IntCounterVec, Registry,
+};
+use std::fmt::Write;
+
+use crate::error::code;
+
+#[derive(Clone)]
+pub(crate) struct Metrics {
+    pub db_metrics: Arc<DBMetrics>,
+    pub request_metrics: Arc<RequestMetrics>,
+}
+
+impl Metrics {
+    pub(crate) fn new(registry: &Registry) -> Self {
+        let db_metrics = DBMetrics::new(&registry);
+        let request_metrics = RequestMetrics::new(&registry);
+
+        Self {
+            db_metrics: Arc::new(db_metrics),
+            request_metrics: Arc::new(request_metrics),
+        }
+    }
+
+    /// Updates the DB related metrics (latency, error, success)
+    pub(crate) fn observe_db_data(&self, db_latency: f64, succeeded: bool) {
+        self.db_metrics
+            .db_fetch_latency
+            .with_label_values(&["latency", "db"])
```

Comment:
This seems redundant, are there other labels we plan to have under the db_fetch_latency metric?

Reply:
This is wrong and I changed it now.